packages/replay-next/components/elements-new/suspense/DOMReactCache.ts

```diff
@@ -0,0 +1,105 @@
+import { ObjectId, PauseId } from "@replayio/protocol";
+import { createCache } from "suspense";
+
+import { objectCache } from "replay-next/src/suspense/ObjectPreviews";
+import { pauseEvaluationsCache } from "replay-next/src/suspense/PauseCache";
+import { findProtocolObjectPropertyValue } from "replay-next/src/utils/protocol";
+import { ReplayClientInterface } from "shared/client/types";
+
+declare var __RECORD_REPLAY__: {
+  getObjectFromProtocolId?: (id: string | number) => any;
+};
+
+declare var __RECORD_REPLAY_ARGUMENTS__: {
+  getPersistentId?: (value: any) => number | null | void;
+};
+
+// TODO [FE-2005][FE-2067] With persistent DOM ids, we could switch this cache from PauseId to execution point.
+
+export type Data = [fiberId: number | null, displayName: string | null];
+
+export const domReactCache = createCache<
+  [replayClient: ReplayClientInterface, pauseId: PauseId, domNodeId: ObjectId],
+  Data
+>({
+  config: { immutable: true },
+  debugLabel: "DOMReactCache",
+  getKey: ([replayClient, pauseId, domNodeId]) => `${pauseId}:${domNodeId}`,
+  load: async ([replayClient, pauseId, domNodeId]) => {
+    const expression = `
+      {
+        const [id, displayName] = (${getReactInfo.toString()})(${domNodeId});
+        JSON.stringify([id, displayName]);
+      }
+    `;
+
+    const result = await pauseEvaluationsCache.readAsync(replayClient, pauseId, null, expression);
+    const value = result?.returned?.value;
+    if (value) {
+      return JSON.parse(value) as Data;
+    }
+
+    // This routine should always return a string value;
+    // if it returns an Object then that is likely an Error.
+    if (result.returned?.object) {
+      const errorObject = await objectCache.readAsync(
+        replayClient,
+        pauseId,
+        result.returned?.object,
+        "full"
+      );
+      const message = findProtocolObjectPropertyValue<string>(errorObject, "message");
+      if (message) {
+        console.error(message);
+      }
+    }
+
+    return [null, null];
+  },
+});
+
+function getReactInfo(domNodeId: ObjectId): Data {
```

Comment:
I wonder if there's a way to flag functions that are meant to be interpolated into an eval? It's a small thing, but it makes reading the code a bit easier.